### ISSUE TITLE
fix(training): six families whose parameter update ignores the target (dead training forward, tape defect, dead training)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#2031]
Six model families fail `TrainingStep_ShouldDependOnTheTarget` for reasons that are **not** the optimizer-reset bugs fixed in the PR that files this issue. Each is a real training defect; they are grouped here because the diagnosis is shared, not the cause.

The invariant asserts that a training step moves parameters in a direction that depends on the target. All six report `cosine 1.000000` — the update is bit-identical for two different targets.

## A. Their training forward returns ALL ZEROS — SegGPT, PixelLM, DiffCutSegmentation, MedSegDiffV2

These four cannot learn at all.

```
SegGPT        ForwardForTraining, training mode : max|v| = 0.000000   (3 consecutive calls)
SegGPT        ForwardForTraining, eval mode     : max|v| = 1.352013E-002
MedSegDiffV2  training mode 0.000000 / eval mode 3.850620E-003
```

Consequence, and it is exact: `BCEWithLogits(z, t) = ln(1+e^z) - t*z`, so at `z = 0` the loss is `ln 2` for **any** target. The models report

```
modelTrainLossA = 6.93147182E-001
modelTrainLossB = 6.93147182E-001    (= ln 2, equal to the last digit)
```

while the correct values at their real prediction are `0.6993` and `0.7009`. A target-invariant loss has a target-invariant gradient, which is why the invariant fires. The target-independence is the SYMPTOM; the disease is a dead training forward.

Ruled out:
- **BatchNorm** — batch=1 training already falls back to running statistics (#639), so it is not the zeroing layer.
- **Gradient checkpointing** — `GradientCheckpointingSegmentSize` defaults to 0, path not taken.
- **The fused compiled path** — all four fail identically with `SupportsFusedCompiledTraining => false`.
- **Optimizer-state carryover** — the PR's two reset fixes reach these models and change nothing.

Note for whoever picks this up: `GetNamedLayerActivations` is NOT a reliable probe here. Its sequential fold falls back to an observed `Predict` when the fold throws, so it reports EVAL activations and shows a healthy non-zero chain while the training forward is zero.

## B. Loss sees the target, the backward does not — Informer

```
modelTrainLossA = 1.74111712E+000
modelTrainLossB = 1.73874223E+000     (different -- the objective DOES see the target)
dA[i] == dB[i] for every sampled i    (identical to 7 significant digits)
```

`predLen = 12288`, plain `MeanSquaredErrorLoss`, no mirroring involved, and `Train` simply delegates to `base.Train`. For MSE the gradient is `2(y - t) * dy/dtheta`, so an update independent of `t` means the `(y, t)` pair reaching the backward is not the pair being scored. A tape defect rather than loss wiring.

Also ruled out: not weight decay. The prediction `delta = -lr*lambda*w` was tested and rejected — `dA/p0` is not constant (`-8.2e-3, -5.2e-2, +5.4e-2, +5.1e-1`) and at one coordinate the delta shares the weight's sign, which decoupled decay never produces.

## C. Dead training — Chronos

Every parameter delta is exactly `0.000000`, and `lossB` reports `-0.0`. Chronos also quantizes the target to token IDs before training (`QuantizeUsingCapturedScale`), so two float targets can collapse onto the same token; that makes an identical update *correct* and means this family needs a target generator aware of its tokenizer before the invariant can measure it at all. Chronos additionally fails 5 tests on master, so it is broadly broken rather than specifically target-independent.

## Status against master

| | Models |
|---|---|
| Failing on master before the PR | MedSegDiffV2, PixelLM, SegGPT (plus TabNet, AdversarialImageEvaluator, AudioVisualEventLocalizationNetwork, LiquidStateMachine — those four are FIXED by the PR) |
| Newly surfaced by the PR | Chronos, DiffCutSegmentation, Informer |

The three newly surfaced ones were passing only because contaminated optimizer state inflated the same-target control, which let the `2x` deficit-ratio margin be met. With the control clean they report honestly. They are pre-existing defects, not regressions introduced by the PR.

## Reproduce

```bash
dotnet test tests/AiDotNet.Tests/AiDotNetTests.csproj --framework net10.0 -c Release \
  --filter "FullyQualifiedName=AiDotNet.Tests.ModelFamilyTests.Generated.SegGPTTests.TrainingStep_ShouldDependOnTheTarget"
```

**Comment by ooples** [comment]
Resolved by #2032 — and the diagnosis in the description above is **wrong** in its main claim, so I am correcting it rather than closing quietly.

## What I got wrong

I reported that SegGPT, PixelLM, DiffCutSegmentation and MedSegDiffV2 have a **dead training forward** and "cannot learn at all", on the evidence that their training loss was exactly `ln 2 = 6.93147182E-001` for both targets, which `BCEWithLogits` only produces at `z = 0`.

That inference was wrong. **SegGPT passes 32 of its 34 tests**, including `Training_ShouldReduceLoss` and `GradientFlow_ShouldBeNonZeroAndFinite`. A model that could not learn would fail those. Informer likewise passed 34 of 36.

## What was actually happening

Three measurement defects in the invariant, not model defects:

1. **The mirror reflected in the wrong space.** The scalar branch builds the second target by reflecting about the prediction so the residual is negated. A `*WithLogits` loss fuses the activation, so its residual is `sigmoid(z) - t`, not `z - t`. On SegGPT (`z = 0.014393`, `tA = 0.068487`) the raw-logit mirror gives residuals `+0.435` and `+0.543` — the same sign, so a **correct** backward returns `cosine +1` and was reported as a defect. Mirroring about `sigmoid(z)` gives `+0.435` and `-0.435`, exactly negated.

2. **Chronos transforms the target before supervising on it** (`QuantizeUsingCapturedScale` to token ids), so two float targets can collapse into the same bucket. `MeasureLoss` compared raw floats and reported a separation of `3.844E+002` for targets the model cannot distinguish.

3. **Informer was under-resolved, not target-blind.** Its cross deficit was `7.261E-011` — non-zero, but below float32 cosine resolution, against a perfectly deterministic control. At 12 steps it separates cleanly and passes.

The `ln 2` reading was real but was a consequence of (1): with the target reflected outside the loss's comparison space the residual never flipped, so the objective could not respond to it.

## Status

All nine families now pass, and a full per-test sweep of every class carrying the invariant is **767 measured, 767 passed, 0 failed**. No follow-up work is outstanding, so this closes rather than staying open as a tracking issue.
